src/ai/defaultai_warfare.cc

```diff
@@ -981,32 +981,71 @@ bool DefaultAI::check_militarysites(const Time& gametime) {
+	// bf.unowned_land_nearby - decreases dismantle probability
+	// portspace_nearby - decreases dismantle probability
+
+	// bf.future_area_military_capacity - increases dismantle probability
+	// bf.military_loneliness - increases dismantle probability
+	// bf.future_military_loneliness - increases dismantle probability
+	// soldier_status_
+	// bf.military_score_ - decreases dismantle probability
+	// bf.own_military_presence - increases dismantle probability
+
+	// bf.enemy_military_presence - decreases dismantle probability
+	// bf.enemy_military_sites - - decreases dismantle probability
+
+	// We calculate the score - if positive, site will be dismounted
+	// We starting with positive number, most modificators are decreasing th score
+	int32_t dismantle_score = 3 * management_data.get_military_number_at(91) / 2 + 100;
+	dismantle_score -= 2 * management_data.neuron_pool[13].get_result_safe(
+	                          bf.enemy_owned_land_nearby / 2, kAbsValue);
+	dismantle_score -= (bf.enemy_owned_land_nearby > 0) ?
+                         std::abs(management_data.get_military_number_at(77)) :
+                         -std::abs(management_data.get_military_number_at(77));
+	dismantle_score -=
+	   management_data.neuron_pool[14].get_result_safe(bf.unowned_land_nearby / 2, kAbsValue);
+	dismantle_score -= (bf.unowned_land_nearby > 0) ?
+                         std::abs(management_data.get_military_number_at(99)) :
+                         -std::abs(management_data.get_military_number_at(99));
+	dismantle_score -= (bf.portspace_nearby == ExtendedBool::kTrue) ?
+                         std::abs(management_data.get_military_number_at(91)) :
+                         0;
+
+	dismantle_score +=
+	   management_data.neuron_pool[28].get_result_safe(bf.area_military_capacity, kAbsValue);
+	dismantle_score +=
+	   management_data.neuron_pool[57].get_result_safe(bf.future_area_military_capacity, kAbsValue);
+	dismantle_score -=
+	   management_data.neuron_pool[29].get_result_safe(bf.military_loneliness / 50, kAbsValue);
+	// dismantle_score -= management_data.neuron_pool[30].get_result_safe( // NOCOM uncomment this
+	//	                        bf.future_military_loneliness, kAbsValue);
+	if (soldier_status_ == SoldiersStatus::kBadShortage) {
+		dismantle_score += 2 * std::abs(management_data.get_military_number_at(77));
+	} else if (soldier_status_ == SoldiersStatus::kShortage) {
+		dismantle_score += std::abs(management_data.get_military_number_at(77));
+	}
+	dismantle_score -=
+	   management_data.neuron_pool[45].get_result_safe(bf.military_score_ / 50, kAbsValue);
+	dismantle_score +=
+	   management_data.neuron_pool[46].get_result_safe(bf.own_military_presence, kAbsValue);
+	dismantle_score -=
+	   management_data.neuron_pool[47].get_result_safe(bf.enemy_military_presence, kAbsValue);
+	dismantle_score -=
+	   management_data.neuron_pool[48].get_result_safe(bf.enemy_military_sites * 3, kAbsValue);
+	dismantle_score -= total_capacity * 2 - 8;
```

Comment:
we should have this dependent on genetics.

How about 
```suggestion
	dismantle_score -= total_capacity * std::abs(management_data.get_military_number_at(89)/20 - std::abs(management_data.get_military_number_at(88)/5;
```

Reply:
After so many changes retraining will be needed so we do not need a replacement for '8', so this would be enough:

dismantle_score -= total_capacity * std::abs(management_data.get_military_number_at(89)/20

Reply:
well I liked the formula as it is discouraging dismantling buildings bigger then x and courages dismantling these who have less then  x. Therefore I would love to keep the part after the minus

Reply:
No, considering that '*' is calculated first and '-' then, that second magic number should not need to be here. Dismantle score is initialized with some value, I would just ignore second magic number - this one would just increase score by some number 0-20. Logically it has no connection to total_capacity.
So I would suggest this:

`dismantle_score -= total_capacity * (1 + std::abs(management_data.get_military_number_at(89)/20);`

Reply:
this would just decrease the value for all buildings. What I liked with the original formula is that small building below the threshhold defined by the part after the minus will be encouraged to dismantle and bigger ones above threshhold will be discouraged from dismantling.

Reply:
But to me this is a duplicity to what is in initialization of the variable:

int32_t dismantle_score = 3 * management_data.get_military_number_at(91) / 2 + 100;

If AI 'thinks' some constant need to be added/deleted to/from the score let it change it here.

Further modification of the score should be really impacts of other variables in the game

Reply:
ok, now I get it. You are right sorry for my misunderstanding.